modules/extension/wmts/src/main/java/org/geotools/data/wmts/model/WMTSCapabilities.java

```diff
@@ -0,0 +1,537 @@
+                            TileMatrixLimits tml = new TileMatrixLimits();
+                            tml.setTileMatix(tmlt.getTileMatrix());
+                            tml.setMinCol(tmlt.getMinTileCol().longValue());
+                            tml.setMaxCol(tmlt.getMaxTileCol().longValue());
+                            tml.setMinRow(tmlt.getMinTileRow().longValue());
+                            tml.setMaxRow(tmlt.getMaxTileRow().longValue());
+                            link.addLimit(tml);
+                        }
+
+                    }
+                    layer.addTileMatrixLink(link);
+                }
+
+                layer.getFormats().addAll(layerType.getFormat());
+
+                layer.getInfoFormats().addAll(layerType.getInfoFormat());
+
+                @SuppressWarnings("unchecked")
+                EList<BoundingBoxType> bboxes = layerType.getBoundingBox();
+                Map<String, CRSEnvelope> boundingBoxes = new HashMap<>();
+                for (BoundingBoxType bbox : bboxes) {
+                    boundingBoxes.put(bbox.getCrs(), bbox2bbox(bbox));
+                }
+
+                WGS84BoundingBoxType wgsBBox = null;
+                if(! layerType .getWGS84BoundingBox().isEmpty()) {
+                    wgsBBox = (WGS84BoundingBoxType) layerType .getWGS84BoundingBox().get(0);
+                }
+                if (wgsBBox != null) {
+                    int y;
+                    int x;
+                    // in WMTS WGS84 is in lon,lat order -
+                    // see https://portal.opengeospatial.org/services/srv_public_issues.php?call=viewIssue&issue_id=898
+                    if (CRS.getAxisOrder(CRS84).equals(AxisDirection.NORTH_EAST)) {
+                        x = 1;
+                        y = 0;
+//                        LOGGER.info("exporting bbox " + wgsBBox + "\n as lat/lon");
+                    } else {
+                        x = 0;
+                        y = 1;
+//                        LOGGER.info("exporting bbox " + wgsBBox + "\n as lon/lat");
+                    }
+                    boundingBoxes.put("CRS:84",
+                            new CRSEnvelope("CRS:84",
+                                    ((Double) wgsBBox.getLowerCorner().get(x)).doubleValue(),
+                                    ((Double) wgsBBox.getLowerCorner().get(y)).doubleValue(),
+                                    ((Double) wgsBBox.getUpperCorner().get(x)).doubleValue(),
+                                    ((Double) wgsBBox.getUpperCorner().get(y)).doubleValue()));
+
+                    layer.setLatLonBoundingBox(boundingBoxes.get("CRS:84"));
+
+
+                }
+                layer.setBoundingBoxes(boundingBoxes);
+
+                EList<URLTemplateType> resourceURL = layerType.getResourceURL();
+                if (resourceURL != null && !resourceURL.isEmpty()) {
+                    for (URLTemplateType resource : resourceURL) {
```

Comment:
"resource" is not used in the loop, resourceURL is instead... does not seem right.